sdk/servicebus/service-bus/test/stress/testRun.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+set -e      # bail if any commands exit with non-zero exit codes.
+
+if [ "$1" == "" ] || [ "$2" == "" ]; then
+    echo "Usage: $0 <scenario.js file (base-name from dist/)> <service-bus-package-version (ex: dev|latest|specific-version)>"
+    exit 1
+fi
+
+# don't forget to fill out the .env file (use sample.env as your base)
+source .env
+
+# figure out the proper image name, based on the git commit
+# and the state of the current clone
+SERVICEBUS_PACKAGE_VERSION=$1
+SCENARIO_JS_BASE_NAME=$2
+KUBERNETES_ENTITY="${SCENARIO_JS_BASE_NAME,,}-$SERVICEBUS_PACKAGE_VERSION"
+
+DOCKER_IMAGE_NAME="$REGISTRY_NAME/$DOCKER_REPOSITORY_NAME:$SERVICEBUS_PACKAGE_VERSION"
+K8S_POD_NAME="$KUBERNETES_ENTITY"
+K8S_SECRET_NAME="$KUBERNETES_ENTITY"
+
+echo "Will deploy"
+echo "  docker image:${DOCKER_IMAGE_NAME}"
+echo "  to pod: ${K8S_POD_NAME}"
+echo "  with secret: ${K8S_SECRET_NAME}"
+echo "  running scenario ${SCENARIO_JS_BASE_NAME}"
+echo "  Using your default context and namespace set in kubectl"
+
+createKubernetesSecret() {
+    echo "Creating Kubernetes secret"
+
+    # create/recreate our secrets.yml
+    SECRETS_YAML=$(<secrets.yml)
+
+    # all the secrets we'll replace (marked with %value% in our secrets.yaml)
+    VARIABLES_TO_INJECT="
+        K8S_SECRET_NAME
+        SERVICEBUS_CONNECTION_STRING
+        APPINSIGHTS_INSTRUMENTATIONKEY
+    "
+
+    for name in $VARIABLES_TO_INJECT; do
+        SECRETS_YAML=${SECRETS_YAML//"%$name%"/${!name}}
+    done
+
+    echo "$SECRETS_YAML" | kubectl replace --force -f -
+}
+
+createKubernetesPod() {
+    echo "Creating Kubernetes pod"
+
+    # recreate the pod
+    POD_YAML=$(<pod.yaml)
+
+    VARIABLES_TO_INJECT="
+        DOCKER_IMAGE_NAME
+        K8S_SECRET_NAME
+        K8S_POD_NAME
+        SCENARIO_JS_BASE_NAME
+    "
+
+    for name in $VARIABLES_TO_INJECT; do
+        POD_YAML=${POD_YAML//"%$name%"/${!name}}
+    done
+
+    echo "$POD_YAML" | kubectl replace --force -f -
```

Comment:
I think an easier approach to injecting the variables, instead of parsing a custom format, is to use envsubst:

```
# file
spec:
  myKey: $SOME_VAR

$ export SOME_VAR=FOO
$ cat <file> | envsubst | kubectl ...
```

Reply:
It can be slightly annoying if some of your manifests actually have a valid `$` character in them, but in practice I've found that to be rare (and there are ways around it).

Reply:
The other benefit of this approach is you can iterate more easily directly from the shell, as you just change your export values and pipe the file to kubectl again (it's also easy to inspect the changes by just excluding the kubectl command).

Reply:
Yeah, I like that suggestion. I'll make a change and push it in a moment.

I wasn't a fan of what I did above (and TBH, I was hoping that there was some kubectl hackery to do this without "custom stuff").

Reply:
Using envsubst!

Reply:
(should note that testRun.sh is minimal enough that it should work well as a 'inner-loop' development tool as well).